QuickBytes/src/main/java/com/quickbytes/controller/RequestController.java

```diff
@@ -0,0 +1,205 @@
+package com.quickbytes.controller;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.quickbytes.enums.RequestStatus;
+import com.quickbytes.model.Customer;
+import com.quickbytes.model.Request;
+import com.quickbytes.model.Vendor;
+import com.quickbytes.repository.CustomerRepository;
+import com.quickbytes.repository.RequestRepository;
+import com.quickbytes.repository.VendorRepository;
+
+@RestController
+public class RequestController {
+	
+	@Autowired
+	private RequestRepository requestRepository;
+	
+	@Autowired
+	private CustomerRepository customerRepository;
+	
+	@Autowired
+	private VendorRepository vendorRepository;
+	
+	/* GET ALL REQUESTS */
+	@GetMapping("/requests")
+	public List<Request> getAllRequests() {
+		return requestRepository.findAll();
+	}
+	
+	/* GET REQUEST BY ID */
+	@GetMapping("/request/{id}")
+	public Request getRequest(@PathVariable("id") Long id) {
+		Optional<Request> request = requestRepository.findById(id);
+		if (request.isPresent()) {
+			return request.get();
+		}
+		throw new RuntimeException("ID is invalid");
+	}
+	
```

Comment:
Same thing as RequestComboController. Apparently having that @PostMapping("/request") followed by the method is important. If necessary, I can ask why.

Reply:
Above